cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -614,7 +614,7 @@ private LocalFile uploadFile(String bucket, String targetFilePath,
         Blob existingBlob = storage.get(blobId);
         BlobInfo newBlobInfo = existingBlob.toBuilder().setContentType(contentType).build();
         uploadToGcsUtil(localFile, storage, targetFilePath, newBlobInfo,
-            Storage.BlobWriteOption.generationNotMatch());
+            Storage.BlobWriteOption.generationMatch());
```

Comment:
What would happen if generationNotMatch is the case when we reach here?

Why not to ignore the exception if generationMatch precondition holds?

Reply:
This logic is becoming a bit complex. I believe we need to add couples of unittests to ensure we correctly cover all cases.

Reply:
Where do we retry?

Reply:
The retry strategy is configured in the storage client: https://github.com/cdapio/cdap/blob/0366675489336099dd8dcbbd5a96a8e0a5159afa/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java#L197-L206

Reply:
Can you add comments regarding why match precondition is being used.

Reply:
I've removed the precondition from second attempt, since we are using `UniformRetryStrategy` which also retries all operations.